lib/src/local_working_copy.rs

```diff
@@ -763,8 +765,34 @@ fn mtime_from_metadata(metadata: &Metadata) -> MillisSinceEpoch {
     )
 }
 
+/// On Windows, checks if a path is a reparse point (junction or symlink).
+/// Junctions have is_dir()=true but is_symlink()=false, so we need this
+/// separate check to avoid recursing into them and hitting "Access denied".
+#[cfg(windows)]
+fn is_reparse_point(metadata: &Metadata) -> bool {
+    const FILE_ATTRIBUTE_REPARSE_POINT: u32 = 0x400;
+    (metadata.file_attributes() & FILE_ATTRIBUTE_REPARSE_POINT) != 0
+}
+
+#[cfg(windows)]
+fn is_normal_dir(entry: &DirEntry) -> Result<bool, SnapshotError> {
+    let metadata = entry.metadata().map_err(|err| SnapshotError::Other {
+        message: format!("Failed to stat file {}", entry.path().display()),
+        err: err.into(),
+    })?;
+    Ok(metadata.is_dir() && !is_reparse_point(&metadata))
+}
+
+#[cfg(not(windows))]
+fn is_normal_dir(entry: &DirEntry) -> Result<bool, SnapshotError> {
+    Ok(entry.file_type().map(|ft| ft.is_dir()).unwrap_or(false))
+}
+
 fn file_state(metadata: &Metadata) -> Option<FileState> {
     let metadata_file_type = metadata.file_type();
+    // On Windows, junctions (created by tools like pnpm) have is_dir()=true
+    // but is_symlink()=false. They cannot be read as symlinks or regular files.
+    // Returning None causes jj to skip them as "special files".
```

Comment:
nit: delete this comment? I don't think this is relevant. Maybe we can move this comment to `is_normal_dir()`?